### PR TITLE
Landing Page - Working With Third-Party Services

### PR DESCRIPTION
**Pull request by aBekar145**

### CODE REVIEW

File: workflow/landing-page/main.js [hunk 1]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+        let response = await fetch(url);
+        let gifData = await response.json();
+        let content = gifData.data;
+        const htmlForRepo = content.map((element) =>
+            createHtmlTemplate(element)
+        );
+        cardsList.innerHTML = '';
+        cardsList.insertAdjacentHTML('afterbegin', htmlForRepo.join(''));
+    } catch (err) {
+        console.log(err, 'ERROR!');
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:52]: use `console.error`/`console.warn` and show error message.
Now you show err object and then ''ERROR!"

File: workflow/landing-page/main.js [hunk 2]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
         navigationIsClosed = true;
     }
 });
+
+async function sendApiRequest(page) {
+    let numberOffset = page === 1 ? 0 : --page * limit;
+    try {
+        const apiKey = 'DDmMIaeRZcddi8NShzyljNpXQ7EOvz6y';
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:41]: take it outside the function. You don't need to create `apiKey` every time the function is called.
And as rule the constants like this named upper cased - `API_KEY`

File: workflow/landing-page/main.js [hunk 3]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+        const url = `https://api.giphy.com/v1/gifs/search?api_key=${apiKey}&q=dogs&limit=${limit}&offset=${numberOffset}`;
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:42]: take it outside and make the function which will take `offset` and return url string

File: workflow/landing-page/main.js [hunk 4]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
         navigationIsClosed = true;
     }
 });
+
+async function sendApiRequest(page) {
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:38]: 1. rename `sendApiRequest` to be more detailed. Now looking at this name I see that this func do any api request. But it gets gifs data and insert them in DOM.
2. you can set default value for function arguments. This value will be used if the argument won't be passed (will be undefined). 
Read this doc [Function Default Arguments](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Functions/Default_parameters) and add default value to sendApiRequest which will let to call `sendApiRequest()` and get the 1st page

File: workflow/landing-page/main.js [hunk 5]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+    let numberOffset = page === 1 ? 0 : --page * limit;
+    try {
+        const apiKey = 'DDmMIaeRZcddi8NShzyljNpXQ7EOvz6y';
+        const url = `https://api.giphy.com/v1/gifs/search?api_key=${apiKey}&q=dogs&limit=${limit}&offset=${numberOffset}`;
+        let response = await fetch(url);
+        let gifData = await response.json();
+        let content = gifData.data;
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:45]: why do you use `let`?

File: workflow/landing-page/main.js [hunk 6]
```diff
@@ -2,9 +2,17 @@ const searchButton = document.getElementById('search-btn');
 const menuButton = document.getElementById('burger-menu-btn');
 const overlayContainer = document.getElementById('overlay');
 const closeOverlayButton = document.getElementById('overlay-close-btn');
+const cardsList = document.getElementById('cards-list');
 const overlayClass = 'closed-overlay-container';
 const responsiveClass = 'closed-responsive-navigation';
 let navigationIsClosed = true;
+const prev = document.getElementById('prev');
+const next = document.getElementById('next');
+const showNumberPage = document.getElementById('page-num');
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:11]: `showNumberPage` looks like name for function which shows smth. Rename it to be smth like `pageNumberElem`

File: workflow/landing-page/main.js [hunk 7]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+
+function check() {
+    if (numberPage == 1) {
+        prev.classList.add('disabled');
+    } else {
+        prev.classList.remove('disabled');
+    }
+}
+
+function createHtmlTemplate(item) {
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:78]: use [destructuring of function arguments](https://davidwalsh.name/destructuring-function-arguments) here to get directly `images`, `url`, `title`, etc

File: workflow/landing-page/main.js [hunk 8]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+                <p class="card__description-text">
+                    Lorem ipsum dolor sit amet, consectetur adipisicing elit.
+                    Alias, deleniti, id quibusdam aut optio saepe soluta tempore neque voluptatum.
+                </p>
+            </div>
+        </div>
+    </li>`;
+}
+
+window.onload = function () {
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:106]: change to arrow func

File: workflow/landing-page/main.js [hunk 9]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+
+function createHtmlTemplate(item) {
+    return `
+    <li class="cards__list-item">
+        <div class="card">
+            <div class="card__container-img">
+                <img class="card__container-img-picture" src="${item.images.downsized.url}" alt="image">
+                <div class="overlay-button-block">
+                    <div class="overlay-button-block__btn">
+                        <a class="fas fa-link" href="${item.url}"></a>
```
Comment (nastystanevich) [inline workflow/landing-page/main.js:86]: use `embed_url` in the item, open in a separate tab

File: workflow/landing-page/stylesheets/style.css [hunk 10]
```diff
@@ -381,9 +381,8 @@ body {
 
 .cards__list {
     display: flex;
-    flex-flow: column wrap;
-    max-height: 1700px;
-    align-content: space-between;
+    flex-wrap: wrap;
+    justify-content: space-between;
 }
```
Comment (nastystanevich) [inline workflow/landing-page/stylesheets/style.css:386]: place items without space in the bottom

image

**Review by nastystanevich — CHANGES_REQUESTED** [review]

File: workflow/landing-page/.vscode/settings.json [hunk 11]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/.vscode/settings.json:2]: done

File: workflow/landing-page/index.html [hunk 12]
```diff
@@ -262,7 +262,7 @@ <h2 class="some-images__title title">
                 </h2>
                 <div class="cards">
                     <div class="cards-wrap wrap">
-                        <ul class="cards__list">
+                        <ul id="delete-cards-list" class="cards__list">
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/index.html:265]: done

File: workflow/landing-page/index.html [hunk 13]
```diff
@@ -482,6 +482,12 @@ <h3 class="card__description-sub-title sub-title">
                                 </div>
                             </li>
                         </ul>
+                        <ul id="cards-list" class="cards__list"></ul>
+                        <div id="pagination"  class="pagination">
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/index.html:486]: done

File: workflow/landing-page/index.html [hunk 14]
```diff
@@ -482,6 +482,12 @@ <h3 class="card__description-sub-title sub-title">
...
+                            <button id="prev" class="pagination__btn btn ">Prev</button>
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/index.html:487]: done

File: workflow/landing-page/index.html [hunk 15]
```diff
@@ -482,6 +482,12 @@ <h3 class="card__description-sub-title sub-title">
...
+                            <div id="page" class="pagination__page">Page <span id="page-num" class="pagination__page-num sub-title"></span></div>
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/index.html:488]: done

File: workflow/landing-page/main.js [hunk 16]
```diff
@@ -2,9 +2,17 @@ const searchButton = document.getElementById('search-btn');
...
 const overlayClass = 'closed-overlay-container';
 const responsiveClass = 'closed-responsive-navigation';
 let navigationIsClosed = true;
+const prev = document.getElementById('prev');
+const next = document.getElementById('next');
+const showNumberPage = document.getElementById('page-num');
+let numberPage = 1;
+const limit = 10;
+const deleteCardsList = document.getElementById('delete-cards-list');
+deleteCardsList.remove();
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/main.js:15]: done

File: workflow/landing-page/main.js [hunk 17]
```diff
@@ -2,9 +2,17 @@ const searchButton = document.getElementById('search-btn');
...
 const closeOverlayButton = document.getElementById('overlay-close-btn');
+const cardsList = document.getElementById('cards-list');
 const overlayClass = 'closed-overlay-container';
 const responsiveClass = 'closed-responsive-navigation';
 let navigationIsClosed = true;
+const prev = document.getElementById('prev');
+const next = document.getElementById('next');
+const showNumberPage = document.getElementById('page-num');
+let numberPage = 1;
+const limit = 10;
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/main.js:13]: done

File: workflow/landing-page/main.js [hunk 18]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
...
+    check();
+});
+
+next.addEventListener('click', function () {
+    numberPage++;
+    sendApiRequest(numberPage);
+    check();
+});
+
+function check() {
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/main.js:70]: done

File: workflow/landing-page/main.js [hunk 19]
```diff
@@ -2,9 +2,17 @@ const searchButton = document.getElementById('search-btn');
...
 let navigationIsClosed = true;
+const prev = document.getElementById('prev');
+const next = document.getElementById('next');
+const showNumberPage = document.getElementById('page-num');
+let numberPage = 1;
+const limit = 10;
+const deleteCardsList = document.getElementById('delete-cards-list');
+deleteCardsList.remove();
 
 function toggleOverlay() {
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/main.js:17]: done

File: workflow/landing-page/main.js [hunk 20]
```diff
@@ -26,3 +34,76 @@ menuButton.addEventListener('click', () => {
         navigationIsClosed = true;
     }
 });
+
+async function sendApiRequest(page) {
+    let numberOffset = page === 1 ? 0 : --page * limit;
```
[the remark this answers is not in the document]
Comment (aBekar145) [reply workflow/landing-page/main.js:39]: done

File: workflow/landing-page/main.js (hunk 2 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:41]: done

File: workflow/landing-page/main.js (hunk 8 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:106]: done

File: workflow/landing-page/main.js (hunk 1 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:52]: done

File: workflow/landing-page/main.js (hunk 5 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:45]: done

File: workflow/landing-page/main.js (hunk 6 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:11]: done

Comment (aBekar145) [reply workflow/landing-page/main.js:11]: done

File: workflow/landing-page/stylesheets/style.css (hunk 10 above)
Comment (aBekar145) [reply workflow/landing-page/stylesheets/style.css:386]: done

File: workflow/landing-page/main.js (hunk 3 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:42]: done

File: workflow/landing-page/main.js (hunk 4 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:38]: done

File: workflow/landing-page/main.js (hunk 9 above)
Comment (aBekar145) [reply workflow/landing-page/main.js:86]: done

File: workflow/landing-page/main.js (hunk 7 above)
Comment (aBekar145) [reply workflow/landing-page/main.js]: done

**Review by nastystanevich — APPROVED** [review]

**Pull request merged**